activerecord/lib/active_record/connection_adapters/sqlite3/schema_dumper.rb

```diff
@@ -5,6 +5,19 @@ module ConnectionAdapters
     module SQLite3
       class SchemaDumper < ConnectionAdapters::SchemaDumper # :nodoc:
         private
+          def virtual_tables(stream)
+            virtual_tables = @connection.virtual_tables
+            if virtual_tables.any?
+              stream.puts "  # Virtual tables defined in this database."
+              stream.puts "  # Note that virtual tables may not work with other database engines. Be careful if changing database."
+              virtual_tables.sort.each do |table_name, options|
+                module_name, arguments = options
+                stream.puts "  create_virtual_table #{table_name.inspect}, #{module_name.inspect}, #{arguments.split(", ").inspect}"
+              end
+              stream.puts
```

Comment:
Minor thing, but since this section is output right before the trailer, the trailing blank line is out of place. It would be nicer to have a blank line preceding. At the moment, the end of `schema.rb` comes out like this:

```ruby
  add_foreign_key "sessions", "users"
  add_foreign_key "webhooks", "users"
  # Virtual tables defined in this database.
  # Note that virtual tables may not work with other database engines. Be careful if changing database.
  create_virtual_table "message_search_index", "fts5", ["body", "tokenize=porter"]

end
```

Reply:
Much nicer - thanks.